webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/domain/DataTypes.scala

```diff
@@ -64,15 +65,24 @@ final case class Manufacturer(name: String) extends AnyVal
  */
 final case class SoftwareEditor(val name: String) extends AnyVal
 
-sealed trait SecurityToken {
-  def key: String
+// this class is used to decode JSON "securityToken" before key validation and typing
+// (the alias is because at some point, it was named "token" in NodeFact)
+final case class JsonDecSecurityToken(@jsonAliases("token") value: String)
+final case class JsonEncSecurityToken(@jsonField("type") @jsonAliases("kind") kind: String, value: String)
+
+sealed abstract class SecurityToken(val kind: String) {
+  def key:   String
+  def value: String
 }
 
 object SecurityToken {
+  implicit val securityTokenEncoder: JsonEncoder[SecurityToken] =
+    DeriveJsonEncoder.gen[JsonEncSecurityToken].contramap(t => JsonEncSecurityToken(t.kind, t.value))
+  implicit val securityTokenDecoder: JsonDecoder[SecurityToken] =
+    DeriveJsonDecoder.gen[JsonDecSecurityToken].mapOrFail(t => Right(Certificate(t.value)))
```

Comment:
Since we can't have anything but `Certificate` since 7.0, it isn't useful to test for the kind, we only have one. If something wasn't migrated since a long time, it should have failing inventories for ages, and the result will "just" be that the content won't be a valide certificate, so generation will fail for that node, which is what we want. 

I didn't do the whole migration that could remove security token trait and only keep certificate and remove a bunch of pattern matcher and etc, because here, it was mostly for the json. 

We can make an other pass of cleaning once we're sure that nothing break here - but it's less important or urgent than moving away from lift-zon.